docs/lazy-adr/adr-004-mvp-light-client.md

```diff
@@ -0,0 +1,273 @@
+Skipping verification only works under the assumption that 2/3+1 of voting power is honest.
+The whole point of doing DAS (and state fraud proofs) is to remove that assumption.
+See also this related issue in the LL specification: [#159](https://github.com/lazyledger/lazyledger-specs/issues/159).
+
+Independent of the existing implementation, there are three ways this could be implemented:
+1. the DAS light client only accepts a header as valid and trusts it after DAS succeeds (additionally to the tendermint verification), and it waits until DAS succeeds (or there was an error or timeout on the way)
+2. (aka 1.5) the DAS light client stages headers where the tendermint verification passes as valid and spins up DAS sampling rotines in the background; the staged headers are committed as valid iff all routines successfully return in time
+3. the DAS light client optimistically accepts a header as valid and trusts it if the regular tendermint verification succeeds; the DAS is run in the background (with potentially much longer timeouts as in 1.) and after the background routine returns (or errs or times out), the already trusted headers are marked as unavailable; this might require rolling back the already trusted headers
+
+We note that from an implementation point of view 1. is not only the simplest approach, but it would also work best with the currently implemented light client design.
+It is the approach that should be implemented first.
+
+The 2. approach can be seen as an optimization where the higher latency DAS can be conducted in parallel for various heights.
+This could speed up catching-up (sequentially) if the light client went offline (shorter than the weak subjectivity time window).
+
+The 3. approach is the most general of all, but it moves the responsibility to wait or to rollback headers to the caller and hence is undesirable as it offers too much flexibility.
+
+
+#### Data Structures
+
+##### LightBlock
+
+As mentioned above the LightBlock should optionally contain the DataAvailabilityHeader.
+```diff
+Index: types/light.go
+===================================================================
+diff --git a/types/light.go b/types/light.go
+--- a/types/light.go	(revision 64fdfaf9143d7af69925aae4f3ae4ea33abd61e5)
++++ b/types/light.go	(date 1620161433642)
+@@ -8,11 +8,15 @@
+ 	tmproto "github.com/lazyledger/lazyledger-core/proto/tendermint/types"
+ )
+
+-// LightBlock is a SignedHeader and a ValidatorSet.
+-// It is the basis of the light client
++// LightBlock is a SignedHeader and a ValidatorSet
++// and optionally a DataAvailabilityHeader (for DAS light clients).
++// It is the basis of the light client.
+ type LightBlock struct {
+-	*SignedHeader `json:"signed_header"`
+-	ValidatorSet  *ValidatorSet `json:"validator_set"`
++	*SignedHeader          `json:"signed_header"`
++	ValidatorSet           *ValidatorSet           `json:"validator_set"`
```

Comment:
These two lines don't seem to actually be changed? Can they be removed from the diff?

Reply:
2dddade

(a lot of `d`s this time)